fdbserver/RestoreMaster.actor.h

```diff
@@ -377,8 +406,17 @@ struct RestoreMasterData : RestoreRoleData, public ReferenceCounted<RestoreMaste
 		// The last wip version batch has some files
 		if (vb.size > 0) {
 			vb.endVersion = std::min(nextVersion, targetVersion + 1);
+			maxVBVersion = std::max(maxVBVersion, vb.endVersion);
 			versionBatches->emplace(vb.beginVersion, vb);
 		}
+		// Invariant: The last vb endverion should be no smaller than targetVersion
+		if (maxVBVersion < targetVersion) {
+			// Q: Is the restorable version always less than the maximum version from all backup filenames?
+			// A: This is true for the raw backup files returned by backup container before we remove the empty files.
```

Comment:
Here I want to validate version batches instead of raw backup files, ensuring every non-empty file is considered.

Because the current implementation filter out the empty files before it starts to construct the version batch, the invariant is not true for version batch analysis.

In the new PR(s), I plan to allow empty files in version batches for validation purpose.